wagtail/admin/localization.py

```diff
@@ -117,3 +122,41 @@ def get_available_admin_time_zones():
     return getattr(
         settings, "WAGTAIL_USER_TIME_ZONES", sorted(zoneinfo.available_timezones())
     )
+
+
+def get_localized_response(view_func, request, *args, **kwargs):
+    user = request.user
+    preferred_language = None
+    if hasattr(user, "wagtail_userprofile"):
+        preferred_language = user.wagtail_userprofile.get_preferred_language()
+        time_zone = user.wagtail_userprofile.get_current_time_zone()
+    else:
+        time_zone = settings.TIME_ZONE
+    with override_tz(time_zone), LogContext(user=user):
```

Comment:
I wonder if `LogContext` should be applied here or kept in `require_admin_access`. The latter would be more appropriate (as it's not related to localization), but the changes wouldn't be a 1:1 cut-paste from the original code.